Automattic/Gravatar-SDK-Android
# QuickEditor: Show currently selected avatar in the list

Closes #256

### Description

This PR uses the `IdentityService` to fetch the selected image id and match it with one in the list of avatars.

I added a new `AvatarRepository` to extract the logic for doing that as it requires 3 separate dependencies. This simplified the ViewModel quite a bit. 

Additionally, I updated the style of the border in the `SelectableAvatar` component to match the designs.


| Light | Dark |
|---|---|
| <img width="580" alt="Screenshot 2024-08-07 at 15 27 30" src="https://github.com/user-attachments/assets/daad1f2c-c709-486f-a2c2-0895c3e2d199"> | <img width="579" alt="Screenshot 2024-08-07 at 15 30 21" src="https://github.com/user-attachments/assets/1b3e5a06-d14a-42b1-98e7-8f859ab8c6aa"> |


### Testing Steps


**Important Note**: The backend is currently returning the wrong format for the `updated date` field. As a workaround, you can apply this patch to comment that field from the Avatar object.
[Gravatar-Android-13-04-48.patch](https://github.com/user-attachments/files/16455284/Gravatar-Android-13-04-48.patch)

1. Make sure you have at least one avatar uploaded on Gravatar
2. Run the DemoApp 
3. Go to Avatar Update tab
4. Tap `Update Avatar` button
5. Follow the steps to log in 
8. Confirm you can see the selected avatar highlighted
